azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/NicIpConfiguration.java

```diff
@@ -80,229 +65,456 @@
+
+            /**
+             * Creates a new virtual network to associate with the network interface Ip configuration.
+             * <p>
+             * the virtual network will be created in the same resource group and region as of parent
+             * network interface, it will be created with the specified address space and a default subnet
+             * covering the entirety of the network Ip address space.
+             *
+             * @param name the name of the new virtual network
+             * @param addressSpace the address space for rhe virtual network
+             * @return the next stage of the network interface Ip configuration definition
+             */
+            WithPrivateIp<ParentT> withNewNetwork(String name, String addressSpace);
+
+            /**
+             * Creates a new virtual network to associate with the network interface Ip configuration.
+             * <p>
+             * the virtual network will be created in the same resource group and region as of parent network interface,
+             * it will be created with the specified address space and a default subnet covering the entirety of the
+             * network IP address space.
+             *
+             * @param addressSpace the address space for the virtual network
+             * @return the next stage of the network interface Ip configuration definition
+             */
+            WithPrivateIp<ParentT> withNewNetwork(String addressSpace);
+
+            /**
+             * Associate an existing virtual network with the network interface Ip configuration.
+             *
+             * @param network an existing virtual network
+             * @return the next stage of the network interface Ip configuration definition
+             */
+            WithSubnet<ParentT> withExistingNetwork(Network network);
+        }
 
         /**
-         * Creates a new virtual network to associate with the Ip configuration.
-         * <p>
-         * the virtual network will be created in the same resource group and region as of parent network interface,
-         * it will be created with the specified address space and a default subnet covering the entirety of the
-         * network IP address space.
+         * The stage of the network interface Ip configuration definition allowing to specify private Ip address
+         * within a virtual network subnet.
          *
-         * @param addressSpace the address space for the virtual network
-         * @return the next stage of the Ip configuration definition
+         * @param <ParentT> the return type of the final {@link Attachable#attach()}
          */
-        DefinitionWithPrivateIp<ParentT> withNewNetwork(String addressSpace);
+        interface WithPrivateIp<ParentT> {
+            /**
+             * Enables dynamic private Ip address allocation within the specified existing virtual network
+             * subnet for the network interface Ip configuration.
```

Comment:
I took care of it in my bigger PR: https://github.com/Azure/azure-sdk-for-java/pull/901